clojure/src/ch01_evaluator.clj

```diff
@@ -223,3 +223,35 @@
 (evaluate '(set! x 1) {})
 ;; => {x 1}
 
+
+;; An empty environment.
+;; Note: in the book they use the name `env.init` but that has a special meaning in Clojure.
+;; It designates a fully-qualified class name such as java.lang.String.
+;; See https://clojure.org/reference/reader#_symbols
+(def env-init {})
+
+
+;; `extend` can enrich environment by assigning variables to their values
+(defn extend [env variables values]
+  ;; we do not yet support special variables like `& args` capturing all the remaining values
+  (if (= (count variables) (count values))
+    (into env (zipmap variables values))
```

Comment:
I was thinking for a while about how to do this easily (again in the book they user recursion) and then remembered the function `zipmap` :).